channelserver/src/main.rs

```diff
@@ -1,357 +1,162 @@
+const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
+/// How long before lack of client response causes a timeout
+const CLIENT_TIMEOUT: Duration = Duration::from_secs(10);
 
 /// Entry point for our route
-fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
-    // not sure if it's possible to have actix_web parse the path and have a properly
-    // scoped request, since the calling structure is different for the two, so
-    // manually extracting the id from the path.
-    let mut path: Vec<_> = req.path().split('/').collect();
-    let meta_info = meta::SenderData::from(req.clone());
-    let mut initial_connect = true;
+async fn channel_route(
+    req: HttpRequest,
+    stream: web::Payload,
+    srv: web::Data<Addr<server::ChannelServer>>,
+) -> Result<HttpResponse, Error> {
+    let raw_state = req.app_data::<web::Data<session::WsChannelSessionState>>();
+    let state = match raw_state {
+        Some(state) => state,
+        None => {
+            return Ok(HttpResponse::InternalServerError().body("Invalid or missing state"));
+        }
+    };
+    let meta = meta::SenderData::new(&req, &state);
+    let mut path: Vec<&str> = req.path().split('/').collect();
+    let log = logging::MozLogger::default();
+    let metrics = state.metrics.clone();
     let channel = match path.pop() {
         Some(id) => {
-            // if the id is valid, but not present, treat it like a "None"
             if id.is_empty() {
                 channelid::ChannelID::default()
             } else {
-                initial_connect = false;
+                // initial_connect = false;
                 match channelid::ChannelID::from_str(id) {
                     Ok(channelid) => channelid,
                     Err(err) => {
-                        warn!(&req.state().log.log,
-                            "Invalid ChannelID specified: {:?}", id;
-                            "remote_ip" => &meta_info.remote);
-                        let mut resp =
-                            HttpResponse::new(http::StatusCode::NOT_FOUND).into_builder();
-                        return Ok(resp.into());
+                        warn!(state.log.log, "Routing error: {:?}", err);
+                        channelid::ChannelID::default()
                     }
                 }
             }
         }
         None => channelid::ChannelID::default(),
     };
-    info!(
-        &req.state().log.log,
-        "Creating session for {} channel: \"{}\"",
-        if initial_connect {"new"} else {"candidate"},
-        channel.to_string();
-        "remote_ip" => &meta_info.remote
-    );
-
-    // Cannot apply headers here.
     ws::start(
-        req,
         session::WsChannelSession {
             id: 0,
             hb: Instant::now(),
-            expiry: Instant::now() + Duration::from_secs(req.state().connection_lifespan),
+            // initial_connect: initial_connect,
```

Comment:
How come we aren't using initial_connect any more? We don't need it?

Reply:
Ah, good catch! Had to fix up something test related and forgot to revert this.